publish-pipeline/content-publish/src/main/scala/org/sunbird/job/content/publish/helpers/ExtractableMimeTypeHelper.scala

```
@@ -0,0 +1,279 @@
+      if (StringUtils.isBlank(ecmlType)) throw new Exception("[System is in a fix between (XML & JSON) ECML Type.]")
+      val file = new File(basePath + "/" + "index." + ecmlType)
+      FileUtils.writeStringToFile(file, ecml, "UTF-8")
+    } catch {
+      case e: Exception =>
+        throw new Exception("[Unable to Write ECML File.]", e)
+    }
+  }
+
+  private def createZipPackage(basePath: String, zipFileName: String): Unit =
+    if (!StringUtils.isBlank(zipFileName)) {
+      logger.info("Creating Zip File: " + zipFileName)
+      val fileList: List[String] = generateFileList(basePath)
+      zipIt(zipFileName, fileList, basePath)
+    }
+
+  private def generateFileList(sourceFolder: String): List[String] =
+    Files.walk(Paths.get(new File(sourceFolder).getPath)).toArray()
+      .map(path => path.asInstanceOf[Path])
+      .filter(path => Files.isRegularFile(path))
+      .map(path => generateZipEntry(path.toString, sourceFolder)).toList
+
+
+  private def generateZipEntry(file: String, sourceFolder: String): String = file.substring(sourceFolder.length, file.length)
+
+  private def zipIt(zipFile: String, fileList: List[String], sourceFolder: String): Unit = {
+    val buffer = new Array[Byte](1024)
+    var zos: ZipOutputStream = null
+    try {
+      zos = new ZipOutputStream(new FileOutputStream(zipFile))
+      logger.info("Creating Zip File: " + zipFile)
+      fileList.foreach(file => {
+        val ze = new ZipEntry(file)
+        zos.putNextEntry(ze)
+        val in = new FileInputStream(sourceFolder + File.separator + file)
+        try {
+          var len = in.read(buffer)
+          while (len > 0) {
+            zos.write(buffer, 0, len)
+            len = in.read(buffer)
+          }
+        } finally if (in != null) in.close()
+        zos.closeEntry()
+      })
+    } catch {
+      case e: IOException => logger.error("Error! Something Went Wrong While Creating the ZIP File: " + e.getMessage, e)
```

We shouldn't suppress the Exception or Failure. Please throw the exception...